lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskImpl.java

```diff
@@ -21,8 +20,8 @@
 public class TaskImpl extends TaskSummaryImpl implements Task {
 
   // All objects have to be serializable
-  private Map<String, String> customAttributes = Collections.emptyMap();
-  private Map<String, String> callbackInfo = Collections.emptyMap();
+  private Map<String, String> customAttributes = new HashMap<>();
+  private Map<String, String> callbackInfo = new HashMap<>();
```

Comment:
Hey Martin,
I know it's a trivial change, but please add a test that verifies that this collection is not immutable anymore. (This also prevents that with the next refactoring we would change it back to immutable).

I would also like to know why this change is required for a private field. Is the actual problem that we cannot modify this map/list? Maybe we should wrap the "add" & "put" option to allow a modification but then just creating a new immutable map and this way keeping the immutability.

Reply:
Yes. Our users want to use `task.getCustomAttributeMap().add(XYZ)`. This is not possible with immutable maps.
There was also a long slack discussion about this.

Reply:
Then this is cleared, but please still add some tests. 

They should be incredibly easy ... 
    `assertThatCode(() -> task.getCustomAttributeMap().add("XYZ")).doesNotThrowAnyException();`

Reply:
:+1:

Reply:
Hi, so I added the test. Is this enough or should I check that every mutable list we have in our entities gets such a test?